IdentityCore/src/webview/background/MSIDBackgroundTaskManager.h

```diff
@@ -0,0 +1,49 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import <Foundation/Foundation.h>
+
+typedef NS_ENUM(NSInteger, MSIDBackgroundTaskType)
+{
+    MSIDInteractiveRequestType = 0
```

Comment:
Why don't we use UIBackgroundTaskIdentifier instead?

Reply:
The caller would need to track background task identifier by themselves then, since background task Identifier gets generated by Apple when we start a task. This is meant to simplify it a bit for the caller. I'm fine either ways, wanted to avoid additional code from caller's perspective :)

Reply:
Makes sense. Nit: can we follow Apple's practice in enum naming? Usually, items of enum are prefixed with enum name.

Reply:
Addressed